### DISCUSSION TITLE
Using beatmap's preview time in the intro

### DISCUSSION THREAD

**Discussion by SK-Fast** [ppy/osu#19687] [Ideas] [answered]
So as right now, the intro(especially triangles intro sequence) is not really good when user disabled osu! music theme in Settings.

So here's my idea: If the osu! music theme is disabled and user is on triangles intro sequence. Use the beatmap's Preview time minus with intro duration, would be cooler than "welcome to osu!"

**Comment by nagi-desuuu** [comment]
I'm having a hard time visualsing this... Can you please elaborate further?
Does the beatmap preview last until the intro sequence is complete? Does it keep playing even after the intro finishes?
And more importantly, would the iconic "welcome to osu" voice still be audible after all that?

**Reply by SK-Fast** [reply]
Beatmap preview is still there until the intro is complete, also keep playing after intro finishes.
I don't think "welcome to osu!" would suit after that.

i'm not so good at explaining, so here's the example video:

https://user-images.githubusercontent.com/60780448/184100343-42bb4dd5-c34d-4b8e-813c-ff64258a9a7b.mp4

**Reply by nagi-desuuu** [reply]
So what I'm seeing here is that the part leading up to the preview point is played during the intro, and the exact preview point comes at the exact moment the intro finishes, then it plays to the end from there.
I kinda like the idea.

**Comment by peppy** [comment, marked as answer]
Check out https://github.com/ppy/osu/pull/19802, which should roughly match what you expect.
